### DISCUSSION TITLE
Huge ./out directory when doing `next export`

### DISCUSSION THREAD

**Discussion by jacdx** [vercel/next.js#12994] [Help]
I recently upgraded to Next 9.4, am only using SSG and my tiny 4-page site is resulting in a 235MB `out` folder. The generated HTMLs and CSS look fine. Well almost, the HTMLs contain large, unnecessary `__NEXT_DATA__` structures which shouldn't be there since the site is exported statically. 

The `_next` folder on the other hand, is full of all sorts of stuff and a lot of it looks redundant, especially the `static/webpack` directory, which has 200MB of hot-update JSONs. There are directoryies like `static/development`, that don't look like they belong in an export. Is there any documentation about what exactly _should_ be in the `_next` folder after an export?

**Comment by mgutz** [comment]
Removing the `.next` directory before `next build` and `next export` brings the size down to a less shocking size. My `out` folder was around 150MB for a simple app. After I removed `.next` before building, the out directory is down to 5MB. Still large. My builds were smaller when I was using a  webpack directly.

**Comment by IRediTOTO** [comment]
My app just 68MB. :\
![image](https://user-images.githubusercontent.com/15695252/86372234-1d49e400-bcac-11ea-88fa-3d2abbda3047.png)

**Comment by guoliu** [comment]
I have the same question. Not only the `out` directory is huge, `.next` directory is also huge, especially `.js` files under `.next/static/chunks/pages`. After building, `Next.js` shows that every page is around 300kb. However several `.js` files under `.next/static/chunks/pages` are more than 15mb.

So my question is,
* How are the page sizes calculated by `Next.js`, and why is it so different from the file size in build directory? `BUNDLE_ANALYZE` looks normal and doesn't include these huge files either.
* Why are these files so big, and how do I build and export a static site with only the code that I need?

**Comment by zaydek** [comment]
This has also been a problem for me when dragging-and-dropping builds to Netlify. Would love to understand if this is intentional behavior (e.g. caching) or a bug. If this is a caching mechanism, is there a way to purge before uploading?

**Comment by jeffscottward** [comment]
Can confirm - a micro site of 3 pages results in 80MB export.
However when trying to export without an existing `.next` folder it asks me to rebuild.
Upon rebuild - the size is brought down 90% - I think the cache just needs to be flushed upon each export.

**Reply by kunKun-tx** [reply]
Can confirm removing `.next` folder reduced `out` folder from 100+MB to 2.2MB when doing static site generation.
Using Nextjs v10.0.5
For static site generation, I only need to do so only once. Could be that the out-of-the-box `.next` folder contains too many backend stuff that got into `out` folder that SSG doesn't need.

**Comment by benvium** [comment]
FYI maintainers are discussing this issue in https://github.com/vercel/next.js/discussions/6009
